ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -218,64 +216,59 @@ public KsqlConfig(Map<?, ?> props) {
     applyStreamsConfig(originals());
   }
 
-  public Map<String, Object> getKsqlConfigProps() {
-    return ksqlConfigProps;
-  }
-
   public Map<String, Object> getKsqlStreamConfigProps() {
-    return ksqlStreamConfigProps;
+    return Collections.unmodifiableMap(ksqlStreamConfigProps);
   }
 
   public Map<String, Object> getKsqlAdminClientConfigProps() {
     final Map<String, Object> adminClientConfigs = new HashMap<>(ksqlStreamConfigProps);
     adminClientConfigs.keySet().retainAll(AdminClientConfig.configNames());
-    return adminClientConfigs;
+    return Collections.unmodifiableMap(adminClientConfigs);
   }
 
-  public Object get(String propertyName) {
-    if (propertyName.toLowerCase().startsWith(KSQL_CONFIG_PROPERTY_PREFIX)) {
-      return ksqlConfigProps.get(propertyName);
-    } else {
-      return ksqlStreamConfigProps.get(propertyName);
-    }
-  }
-
-  public void put(String propertyName, Object propertyValue) {
-    if (propertyName.toLowerCase().startsWith(KSQL_CONFIG_PROPERTY_PREFIX)) {
-      ksqlConfigProps.put(propertyName, propertyValue);
-    } else if (propertyName.startsWith(KSQL_STREAMS_PREFIX)) {
-      ksqlStreamConfigProps.put(
-          propertyName.substring(KSQL_STREAMS_PREFIX.length()), propertyValue);
-    } else {
-      ksqlStreamConfigProps.put(propertyName, propertyValue);
-    }
+  public Map<String, Object> getAllProps() {
+    final Map<String, Object> props = new HashMap<>();
+    props.putAll(values().entrySet().stream()
+        .filter(e -> e.getValue() != null)
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue))
+    );
+    props.putAll(ksqlStreamConfigProps.entrySet().stream()
+        .filter(e -> e.getValue() != null)
+        .collect(
+            Collectors.toMap(
+                e -> KSQL_STREAMS_PREFIX + e.getKey(),
+                Map.Entry::getValue)));
+    return Collections.unmodifiableMap(props);
   }
 
   public KsqlConfig clone() {
-    Map<String, Object> clonedProperties = new HashMap<>();
+    final Map<String, Object> clonedProperties = new HashMap<>();
     clonedProperties.putAll(originals());
     return new KsqlConfig(clonedProperties);
   }
 
   public KsqlConfig cloneWithPropertyOverwrite(final Map<String, Object> props) {
-    final Map<String, Object> cloned = new HashMap<>();
+    final Map<String, Object> cloneProps = new HashMap<>(originals());
+    cloneProps.putAll(props);
+    final KsqlConfig clone = new KsqlConfig(cloneProps);
+    clone.applyStreamsConfig(props);
```

Comment:
Isn't this superfluous, given its called inside the `KsqlConfig` constructor?

Reply:
It ensures the overwriting streams properties take precedence in case the originals had un-prefixed properties and the overwrite doesn't (or vice-versa)